### ISSUE TITLE
SQLLite3.OperationalError + OSError

### ISSUE THREAD

**Issue by Lark-3301** [smicallef/spiderfoot#1759]
Hi,

I was running a long scan and spiderfoot suddenly stop scanning while it prompts this errors:

<img width="1208" alt="Error SP-Edited" src="https://user-images.githubusercontent.com/69369181/220631491-bf128527-cbcf-4f4f-8ed3-502c205ac224.png">

It seems a problem of space but I have 20 GB of free space on my VM. Maybe the DB whit which operates spiderfoot is limited and I exceeded its capacity?. But the "OSError" puzzles me.

I had to uninstall Spiderfoot and reinstall it. Now is working well but... do you know any solution not to make this tedious process?

Than you in advance.

**Comment by bcoles** [comment]
> It seems a problem of space but I have 20 GB of free space on my VM. 

Are you using multiple partitions? Logs are stored in the user's home directory by default.

> But the "OSError" puzzles me.

The `SpiderFootDb` class raises `OSError`. This is hardcoded and perhaps not the best error type.

**Comment by vicmac-github** [comment]
> @bcoles 

I have only one partition. That´s why it is strange..

Do you know any solution?

Thank you in advance!

**Comment by bcoles** [comment]
> Do you know any solution?

I don't understand. You said you have it working now? :

> I had to uninstall Spiderfoot and reinstall it. Now is working well but... do you know any solution not to make this tedious process?

**Comment by vicmac-github** [comment]
@bcoles 

Yes but it brokes up again and again and I have to uninstall and reinstall againd and it´s exasperating. It´s like if the DB doesn´t support a huge scanning. I don´t know what´s going on... My last scan stops with that error when it achieve 28K elements.

**Comment by bcoles** [comment]
> Yes but it brokes up again and again and I have to uninstall and reinstall again

That doesn't sounds right. Uninstalling and reinstalling should make no difference.

Are you sure you disk is not full?

SpiderFoot opens two copies of the database. The second copy is removed from disk when SpiderFoot is closed.

Try checking your available disk space with `df -h` while SpiderFoot is running.

**Comment by vicmac-github** [comment]
@bcoles 

this is my disk memory

<img width="460" alt="Kali Linux - Disk Space" src="https://user-images.githubusercontent.com/69369181/220711413-a9e267b9-bb4c-48ac-b6e9-938d2d17a5e2.png">

**Comment by bcoles** [comment]
> this is my disk memory

Your disk is almost full.
